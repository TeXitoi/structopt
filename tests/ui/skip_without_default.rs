// Copyright 2018 Guillaume Pinot (@TeXitoi) <texitoi@texitoi.eu>
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use structopt::StructOpt;

#[derive(Debug)]
enum Kind {
    A,
    B,
}

#[derive(StructOpt, Debug)]
#[structopt(name = "test")]
pub struct Opt {
    #[structopt(short)]
    number: u32,
    #[structopt(skip)]
    k: Kind,
}

fn main() {
    let opt = Opt::from_args();
    println!("{:?}", opt);
}
