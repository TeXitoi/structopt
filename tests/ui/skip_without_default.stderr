error[E0277]: the trait bound `Kind: Default` is not satisfied
  --> $DIR/skip_without_default.rs:22:17
   |
22 |     #[structopt(skip)]
   |                 ^^^^ the trait `Default` is not implemented for `Kind`
   |
   = note: required by `std::default::Default::default`
